### PR TITLE
Specific file types in WS

### PR DESCRIPTION
**Pull request by dimitarquanterall**
Signed-off-by: dimitarquanterall <d.aleksandrov@sap.com>

### What does this PR do?

Adds support for new options in extension definitions - nameless, one file of a type per folder, static name (create without editing name)

### What issues does this PR fix or reference?

https://github.com/SAP/xsk/issues/765

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -71,7 +71,7 @@
 	<description>Dirigible Parent</description>
 	<groupId>org.eclipse.dirigible</groupId>
 	<artifactId>dirigible-parent</artifactId>
-	<version>7.0.0-SNAPSHOT</version>
+	<version>7.0.0-cd </version>
```
Comment (StanZGenchev) [inline pom.xml:74]: This is wrong. The version here should not be edited manually.

**Review by tdermendjiev — APPROVED** [review]

File: ide/ui/ide-workspace/src/main/resources/META-INF/dirigible/ide-workspace/workspace.js [hunk 2]
```diff
@@ -1097,46 +1107,104 @@ angular.module('workspace', ['workspace.config', 'ideUiCore', 'ngAnimate', 'ngSa
...
+                                        type: 'file',
+                                        data: fileTemplate.data,
                                     };
-                                    fileNode.text = 'file.' + fileTemplate.extension;
-                                    fileNode.data = fileTemplate.data;
-                                    tree.create_node(parentNode, fileNode, "last", function (new_node) {
-                                        tree.edit(new_node);
-                                    });
+                                    if (fileTemplate.nameless) fileNode.text = `.${fileTemplate.extension}`;
+                                    else fileNode.text = `file.${fileTemplate.extension}`;
```
Comment (ThuF) [inline ide/ui/ide-workspace/src/main/resources/META-INF/dirigible/ide-workspace/workspace.js:1144]: Please do `if-else-if-else` statements (even single lined), in the following way:
```javascript
if (fileTemplate.nameless) {
  fileNode.text = `.${fileTemplate.extension}`;
} else {
  fileNode.text = `file.${fileTemplate.extension}`;
}
```

File: ide/ui/ide-workspace/src/main/resources/META-INF/dirigible/ide-workspace/workspace.js [hunk 3]
```diff
@@ -1097,46 +1107,104 @@ angular.module('workspace', ['workspace.config', 'ideUiCore', 'ngAnimate', 'ngSa
...
+                                for (let nc = 0; nc < nodeChildren.length; nc++) {
+                                    if (fileTemplate.nameless) {
+                                        if (nodeChildren[nc] === `.${fileTemplate.extension}`) {
+                                            isDisabled = true;
+                                            break;
+                                        }
+                                    } else if (nodeChildren[nc].endsWith(`.${fileTemplate.extension}`)) {
+                                        isDisabled = true;
+                                        break;
+                                    }
```
Comment (ThuF) [inline ide/ui/ide-workspace/src/main/resources/META-INF/dirigible/ide-workspace/workspace.js:1176]: The following code can be refactored as follows:
```javascript
if (fileTemplate.nameless) {
  if (nodeChildren[nc] === `.${fileTemplate.extension}`) {
    isDisabled = true;
    break;
  }
} else if (nodeChildren[nc].endsWith(`.${fileTemplate.extension}`)) {
  isDisabled = true;
  break;
}
```
&#8594;
```javascript
if (fileTemplate.nameless && nodeChildren[nc] === `.${fileTemplate.extension}` || nodeChildren[nc].endsWith(`.${fileTemplate.extension}`)) {
  isDisabled = true;
  break;
}
```

File: ide/ui/ide-workspace/src/main/resources/META-INF/dirigible/ide-workspace/workspace.js [hunk 4]
```diff
@@ -1097,46 +1107,104 @@ angular.module('workspace', ['workspace.config', 'ideUiCore', 'ngAnimate', 'ngSa
...
+                                        type: 'file',
+                                        data: fileTemplate.data
                                     };
-                                    fileNode.text = 'file.' + fileTemplate.extension;
-                                    fileNode.data = fileTemplate.data;
-                                    tree.create_node(parentNode, fileNode, "last", function (new_node) {
-                                        tree.edit(new_node);
-                                    });
+                                    if (fileTemplate.nameless) fileNode.text = `.${fileTemplate.extension}`;
+                                    else fileNode.text = `file.${fileTemplate.extension}`;
```
Comment (ThuF) [inline ide/ui/ide-workspace/src/main/resources/META-INF/dirigible/ide-workspace/workspace.js:1190]: Same about the `if-else-if-else`

File: ide/ui/ide-workspace/src/main/resources/META-INF/dirigible/ide-workspace/workspace.js [hunk 5]
```diff
@@ -1363,15 +1431,13 @@ angular.module('workspace', ['workspace.config', 'ideUiCore', 'ngAnimate', 'ngSa
         uploader.filters.push({
             name: 'customFilter',
             fn: function (item /*{File|FileLikeObject}*/, options) {
+                let type = item.type.slice(item.type.lastIndexOf('/') + 1);
+                if (type != 'zip' && type != 'x-zip') return false;
```
Comment (ThuF) [inline ide/ui/ide-workspace/src/main/resources/META-INF/dirigible/ide-workspace/workspace.js:1435]: Same about `if-else-if-else`

File: ide/ui/ide-workspace/src/main/resources/META-INF/dirigible/ide-workspace/workspace.js [hunk 6]
```diff
@@ -1097,46 +1107,104 @@ angular.module('workspace', ['workspace.config', 'ideUiCore', 'ngAnimate', 'ngSa
...
+                                for (let nc = 0; nc < nodeChildren.length; nc++) {
+                                    if (fileTemplate.nameless) {
+                                        if (nodeChildren[nc] === `.${fileTemplate.extension}`) {
+                                            isDisabled = true;
+                                            break;
+                                        }
+                                    } else if (nodeChildren[nc].endsWith(`.${fileTemplate.extension}`)) {
+                                        isDisabled = true;
+                                        break;
+                                    }
```
Comment (ThuF) [inline ide/ui/ide-workspace/src/main/resources/META-INF/dirigible/ide-workspace/workspace.js:1129]: Please reduce the complexity, as shown bellow.

File: ide/ui/ide-workspace/src/main/resources/META-INF/dirigible/ide-workspace/workspace.js [hunk 7]
```diff
@@ -1015,6 +1015,15 @@ angular.module('workspace', ['workspace.config', 'ideUiCore', 'ngAnimate', 'ngSa
 
         let priorityFileTemplates = JSON.parse(templates).filter(e => e.order !== undefined).sort((a, b) => a.order - b.order);
         let specificFileTemplates = JSON.parse(templates).filter(e => e.order === undefined);
+        let post_no_edit_URL = '/services/v4/ide/workspaces/';
+
+        for (let i = 0; i < priorityFileTemplates.length; i++) {
+            if (priorityFileTemplates[i].isModel) models.push(priorityFileTemplates[i].extension);
```
Comment (ThuF) [inline ide/ui/ide-workspace/src/main/resources/META-INF/dirigible/ide-workspace/workspace.js:1021]: Same about `if-else-if-else`

File: ide/ui/ide-documents/src/main/resources/META-INF/dirigible/ide-documents/index.html [hunk 8]
```diff
@@ -68,16 +68,30 @@
...
+                    $scope.layoutModel.viewSettings['documents-constraints'] = { isClosable: false };
+                })
+                .catch(function(err) {
+                    //User is not Operator, so Constraints view cannot be presented."
+                });
+
+                function getConstraints() {
+                    let requestUrl = '/services/v4/js/ide-documents/api/constraints';
+
+                    return $http.get(requestUrl);
```
Comment (ThuF) [inline ide/ui/ide-documents/src/main/resources/META-INF/dirigible/ide-documents/index.html:92]:
```javascript
return $http.get('/services/v4/js/ide-documents/api/constraints');
```

**Review by ThuF — CHANGES_REQUESTED** [review]

**Review by StanZGenchev — APPROVED** [review]

**Review by ThuF — APPROVED** [review]

**Pull request merged**